2020/BillSplitter/BillSplitter/Data/OrdersDbHelper.cs

```diff
@@ -17,17 +19,17 @@ public void AddOrder(Order order)
             _context.SaveChanges();
         }
 
-        public void AddOrders(Customer customer, int[] selected, int[] numerator, int[] denomenator)
+        public void AddOrders(Customer customer, List<InteractionLevelPosition> positions)
         {
-            for (int i = 0; i < selected.Length; i++)
+            foreach (var pos in positions)
             {
-                if (1.0 * numerator[i] / denomenator[i] > double.Epsilon)
+                if (pos.Selected && 1.0 * pos.QuantityNumerator / pos.QuantityDenomenator > double.Epsilon)
```

Comment:
AddOrders looks functional complicated. I think there is a need to transfer "positions" with filtering by "Selected" and validating by Quantity params. Or sth like that.

Reply:
Well. This can be tranformed into one linq query in AddOrders, or into two linq queries: one for filtering by Selected before calling AddOrders and one just for contruction orders in this method.

Reply:
Fixed with second solution. Check latest commit.